[BUG] scala-sttp string interpolation with backticks compiler error

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ 5.0.0-beta] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

If REST API method parameter is called `type` with backticks, then Scala compiler fails with:

```
nifi-client/target/generated-sources/openapi/src/gen/scala/main/org/openapitools/client/api/FlowApi.scala:569: error: invalid string interpolation $`, expected: $$, $identifier or ${expression}
[ERROR]       .method(Method.GET, uri"$baseUrl/flow/processor-types?bundleGroupFilter=$bundleGroupFilter&bundleArtifactFilter=$bundleArtifactFilter&type=$`type`")
```
Example is taken from NiFi REST API: https://nifi.apache.org/docs/nifi-docs/rest-api/index.html

Expected behavior: code compiles without an y error.

##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->

##### Command line used for generation

mvn clean install

##### Steps to reproduce

1. Use Maven plugin:
```
<groupId>org.openapitools</groupId>
<artifactId>openapi-generator-maven-plugin</artifactId>
<version>5.0.0-beta</version>
```
2. Specify generator as:
```
<generatorName>scala-sttp</generatorName>
```
3. Specify NiFi swagger JSON file
4. Run `mvn clean install`

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

Possible solution: wrap Scala keyword like name, i.e. `type` with curly braces. i.e. generated string must be like this:

```scala
.method(Method.GET, uri"$baseUrl/flow/processor-types?bundleGroupFilter=$bundleGroupFilter&bundleArtifactFilter=$bundleArtifactFilter&type=${`type`}")
```

CC @ghostbuster91

--- a/modules/openapi-generator/src/main/resources/scala-sttp/api.mustache
+++ b/modules/openapi-generator/src/main/resources/scala-sttp/api.mustache
@@ -22,7 +22,7 @@ class {{classname}}(baseUrl: String) {
 {{/javadocRenderer}}
   def {{operationId}}({{>methodParameters}}): Request[{{#separateErrorChannel}}Either[ResponseError[Exception], {{>operationReturnType}}]{{/separateErrorChannel}}{{^separateErrorChannel}}{{>operationReturnType}}{{/separateErrorChannel}}, Nothing] =
     basicRequest
-      .method(Method.{{httpMethod.toUpperCase}}, uri"$baseUrl{{{path}}}{{#queryParams.0}}?{{#queryParams}}{{baseName}}=${{{paramName}}}{{^-last}}&{{/-last}}{{/queryParams}}{{/queryParams.0}}{{#isApiKey}}{{#isKeyInQuery}}{{^queryParams.0}}?{{/queryParams.0}}{{#queryParams.0}}&{{/queryParams.0}}{{keyParamName}}=${apiKey.value}&{{/isKeyInQuery}}{{/isApiKey}}")
+      .method(Method.{{httpMethod.toUpperCase}}, uri"$baseUrl{{{path}}}{{#queryParams.0}}?{{#queryParams}}{{baseName}}=${ {{{paramName}}} }{{^-last}}&{{/-last}}{{/queryParams}}{{/queryParams.0}}{{#isApiKey}}{{#isKeyInQuery}}{{^queryParams.0}}?{{/queryParams.0}}{{#queryParams.0}}&{{/queryParams.0}}{{keyParamName}}=${apiKey.value}&{{/isKeyInQuery}}{{/isApiKey}}")
       .contentType({{#consumes.0}}"{{{mediaType}}}"{{/consumes.0}}{{^consumes}}"application/json"{{/consumes}}){{#headerParams}}
       .header({{>paramCreation}}){{/headerParams}}{{#authMethods}}{{#isBasic}}{{#isBasicBasic}}
       .auth.withCredentials(username, password){{/isBasicBasic}}{{#isBasicBearer}}

--- a/modules/openapi-generator/src/main/resources/scala-sttp/paramQueryCreation.mustache
+++ b/modules/openapi-generator/src/main/resources/scala-sttp/paramQueryCreation.mustache
@@ -1,1 +1,0 @@
-{{#isContainer}}${ formatQueryArray("{{{baseName}}}",{{{paramName}}}{{#collectionFormat}}, {{collectionFormat.toUpperCase}}{{/collectionFormat}}) }{{/isContainer}}{{^isContainer}}{{baseName}}=${ {{{paramName}}} }{{/isContainer}}

--- a/samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala
+++ b/samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala
@@ -67,7 +67,7 @@ class PetApi(baseUrl: String) {
   def findPetsByStatus(status: Seq[String]
 ): Request[Either[ResponseError[Exception], Seq[Pet]], Nothing] =
     basicRequest
-      .method(Method.GET, uri"$baseUrl/pet/findByStatus?status=$status")
+      .method(Method.GET, uri"$baseUrl/pet/findByStatus?status=${ status }")
       .contentType("application/json")
       .response(asJson[Seq[Pet]])
 
@@ -83,7 +83,7 @@ class PetApi(baseUrl: String) {
   def findPetsByTags(tags: Seq[String]
 ): Request[Either[ResponseError[Exception], Seq[Pet]], Nothing] =
     basicRequest
-      .method(Method.GET, uri"$baseUrl/pet/findByTags?tags=$tags")
+      .method(Method.GET, uri"$baseUrl/pet/findByTags?tags=${ tags }")
       .contentType("application/json")
       .response(asJson[Seq[Pet]])
 

--- a/samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/UserApi.scala
+++ b/samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/UserApi.scala
@@ -129,7 +129,7 @@ class UserApi(baseUrl: String) {
   def loginUser(username: String, password: String
 ): Request[Either[ResponseError[Exception], String], Nothing] =
     basicRequest
-      .method(Method.GET, uri"$baseUrl/user/login?username=$username&password=$password")
+      .method(Method.GET, uri"$baseUrl/user/login?username=${ username }&password=${ password }")
       .contentType("application/json")
       .response(asJson[String])
 

